### PR TITLE
Refactor1 : Worked on the failed build for previous PR #739

### PR DESCRIPTION
**Pull request by almasfiza**
The build was failing because of using the same function name for int and long generating functions. I have only added the code for int function - decomposed the conditional in refactoring, all the rest of the changes are as same as previous PR. (https://github.com/datafaker-net/datafaker/pull/739)

### CODE REVIEW

File: src/main/java/net/datafaker/providers/base/Number.java [hunk 1]
```diff
@@ -74,6 +78,7 @@ public long numberBetween(long min, long max) {
         return decimalBetween(realMin, realMax).longValue();
     }
 
+
```
Comment (snuyanzin) [inline src/main/java/net/datafaker/providers/base/Number.java:81]: why do we need this line?

File: src/main/java/net/datafaker/service/FakerIDN.java [hunk 2]
```diff
@@ -17,16 +17,16 @@ public static String toASCII(String in) {
         } catch (Exception e) {
             // let's continue with the character by character encoding hack.
         }
-        final StringBuilder sb = new StringBuilder();
+        final StringBuilder resultASCII = new StringBuilder();
```
Comment (kingthorin) [inline src/main/java/net/datafaker/service/FakerIDN.java:20]: I'd suggest `resultAscii` to actually follow Java naming conventions.

**Comment by kingthorin** [comment]
The failure must be related to your Number.java changes.

```
2023-03-29T02:31:45.1267352Z Caused by: java.lang.IllegalArgumentException: bound must be positive
2023-03-29T02:31:45.1267874Z 	at net.datafaker.service.RandomService.nextLong(RandomService.java:50)
2023-03-29T02:31:45.1268426Z 	at net.datafaker.providers.base.Number.numberBetween(Number.java:79)
2023-03-29T02:31:45.1269106Z 	at net.datafaker.providers.base.NumberTest.lambda$numberBetweenLongLongRandomizationQuality$4(NumberTest.java:255)
2023-03-29T02:31:45.1269860Z 	at net.datafaker.providers.base.NumberTest.uniquePercentageOfResults(NumberTest.java:326)
2023-03-29T02:31:45.1270341Z 	... 49 more
```

**Comment by kingthorin** [comment]
@snuyanzin is the JavaDoc in Number.java wrong?

```java
     * @param min the lower bound (include min)
     * @param max the lower bound (not include max)
```

Shouldn't max be "upper"?

**Comment by snuyanzin** [comment]
>@snuyanzin is the JavaDoc in Number.java wrong?

yes, 3 times
good catch

File: src/main/java/net/datafaker/providers/base/Number.java (hunk 1 above)
Comment (almasfiza) [reply src/main/java/net/datafaker/providers/base/Number.java:81]: > why do we need this line?

Should I just call, isValidRange() and also make a a function isValidRange(long, long)? I tried doing it but the build failed.

File: src/main/java/net/datafaker/providers/base/Number.java [hunk 3]
```diff
@@ -52,11 +52,15 @@ public int numberBetween(int min, int max) {
...
         final int realMin = Math.min(min, max);
         final int realMax = Math.max(min, max);
-        if (realMax - realMin >= realMin && (realMin >= 0 || realMax - realMin >= realMax)) {
+        if (isValidRange(realMin, realMax)) {
             return faker.random().nextInt(realMax - realMin) + realMin;
         }
         return decimalBetween(realMin, realMax).intValue();
     }
+    private boolean isValidRange(int realMin, int realMax) {
+        int amplitude = realMax - realMin;
```
[the remark this answers is not in the document]
Comment (almasfiza) [reply src/main/java/net/datafaker/providers/base/Number.java:61]: > this should be `final`

fixed.

File: src/main/java/net/datafaker/providers/base/Number.java (hunk 1 above)
Comment (snuyanzin) [reply src/main/java/net/datafaker/providers/base/Number.java:81]: >I tried doing it but the build failed.

in fact you didn't . If you look at your old changes there was a method for `float` but not for `long` that's the problem
https://github.com/datafaker-net/datafaker/pull/739/files#diff-94330e19cbdfbd84d55b982b84194e54d5a83486d1c78ee333fcaa76b5238ac3R84

File: src/main/java/net/datafaker/service/FakerIDN.java (hunk 2 above)
Comment (almasfiza) [reply src/main/java/net/datafaker/service/FakerIDN.java]: > I'd suggest `resultAscii` to actually follow Java naming conventions.

fixed.

File: src/main/java/net/datafaker/providers/base/Number.java (hunk 1 above)
Comment (almasfiza) [reply src/main/java/net/datafaker/providers/base/Number.java:86]: > > I tried doing it but the build failed.
> 
> in fact you didn't . If you look at your old changes there was a method for `float` but not for `long` that's the problem https://github.com/datafaker-net/datafaker/pull/739/files#diff-94330e19cbdfbd84d55b982b84194e54d5a83486d1c78ee333fcaa76b5238ac3R84
oh yes, i completely missed it! thank you for looking into it. fixed.

**Comment by almasfiza** [comment]
> The failure must be related to your Number.java changes.
> 
> ```
> 2023-03-29T02:31:45.1267352Z Caused by: java.lang.IllegalArgumentException: bound must be positive
> 2023-03-29T02:31:45.1267874Z 	at net.datafaker.service.RandomService.nextLong(RandomService.java:50)
> 2023-03-29T02:31:45.1268426Z 	at net.datafaker.providers.base.Number.numberBetween(Number.java:79)
> 2023-03-29T02:31:45.1269106Z 	at net.datafaker.providers.base.NumberTest.lambda$numberBetweenLongLongRandomizationQuality$4(NumberTest.java:255)
> 2023-03-29T02:31:45.1269860Z 	at net.datafaker.providers.base.NumberTest.uniquePercentageOfResults(NumberTest.java:326)
> 2023-03-29T02:31:45.1270341Z 	... 49 more
> ```

Yes, thank you for pointing it out, fixed.

**Comment by almasfiza** [comment]
> @snuyanzin is the JavaDoc in Number.java wrong?
> 
> ```java
>      * @param min the lower bound (include min)
>      * @param max the lower bound (not include max)
> ```
> 
> Shouldn't max be "upper"?

yes! did not notice it at first, made the changes in java doc.

**Comment by kingthorin** [comment]
Thanks

**Comment by kingthorin** [comment]
They haven't posted an [outage](https://www.githubstatus.com/) yet but it seems like something is up with GitHub ations.

**Comment by kingthorin** [comment]
https://www.githubstatus.com/incidents/z3c6q056q332

**Comment by almasfiza** [comment]
> https://www.githubstatus.com/incidents/z3c6q056q332

image
Should I work on something from my side for this? I tried doing maven test and compile on my local and it works.

**Comment by kingthorin** [comment]
I think you need to pickup these changes: https://github.com/datafaker-net/datafaker/pull/745

Assuming `upstream` is this repo, `origin` is your fork and you have the PR branch checked out:

```bash
git fetch upstream
git rebase upstream/main
git push --force-with-lease
```

**Comment by almasfiza** [comment]
> I think you need to pickup these changes: #745
> 
> Assuming `upstream` is this repo, `origin` is your fork and you have the PR branch checked out:
> 
> ```shell
> git fetch upstream
> git rebase upstream/main
> git push --force-with-lease
> ```

am i supposed to write https://github.com/datafaker-net/datafaker.git instead of upstream? I am getting an issue with my upstream.
image

**Comment by almasfiza** [comment]
> I think you need to pickup these changes: #745
> 
> Assuming `upstream` is this repo, `origin` is your fork and you have the PR branch checked out:
> 
> ```shell
> git fetch upstream
> git rebase upstream/main
> git push --force-with-lease
> ```

image

I get this error.

**Comment by almasfiza** [comment]
> Or maybe you aren't tracking the right repo. Try `push -u origin refactor1 --force-with-lease` if that fails then `git remote -v` and provide the output.

Yes, i tried changing the origin to my repo, but now when i try fetch, it does not return anything.

image

**Comment by kingthorin** [comment]
Okay, when you fetch now there's nothing to fetch because you've done it and things are up-to-date. Have you made local changes that haven't been pushed? What does `git status` say?

**Comment by almasfiza** [comment]
> Okay, when you fetch now there's nothing to fetch because you've done it and things are up-to-date. Have you made local changes that haven't been pushed? What does `git status` say?

I have not made any changes. git status shows On branch refactor1 nothing to commit, working tree clean

**Comment by kingthorin** [comment]
Okay, try this then.

```bash
git fetch origin
# Shouldn't really do anything but maybe it'll sort out the stale situation
git fetch upstream
# Shouldn't really do anything because there hasn't been any changes in the last hour or so, but may as well ensure up-to-date
git rebase upstream/main
git push -u origin refactor1 --force
```

**Comment by kingthorin** [comment]
Yay, looks like it worked out that time 🎉

**Comment by almasfiza** [comment]
> Yay, looks like it worked out that time 🎉

Yes, it works! Hoping it passes all the checks. Thank you very much for your help and patience! :') Still figuring open source out.

**Comment by kingthorin** [comment]
No problem, there's a learning curve, been there :wink: all good.

File: src/main/java/net/datafaker/providers/base/Number.java [hunk 4]
```diff
@@ -44,35 +44,45 @@ public int negative() {
...
             return faker.random().nextLong(realMax - realMin) + realMin;
         }
         return decimalBetween(realMin, realMax).longValue();
     }
+    private boolean isValidRange(long realMin, long realMax) {
+        final long amplitude = realMax - realMin;
+        return amplitude >= realMin && (realMin >= 0 || amplitude >= realMax);
+    }
+
+
```
Comment (snuyanzin) [inline src/main/java/net/datafaker/providers/base/Number.java:86]: between methods there should be only one empty line

File: src/main/java/net/datafaker/providers/base/Number.java [hunk 5]
```diff
@@ -44,35 +44,45 @@ public int negative() {
...
         if (min == max) return min;
         final int realMin = Math.min(min, max);
         final int realMax = Math.max(min, max);
-        if (realMax - realMin >= realMin && (realMin >= 0 || realMax - realMin >= realMax)) {
+        if (isValidRange(realMin, realMax)) {
             return faker.random().nextInt(realMax - realMin) + realMin;
         }
         return decimalBetween(realMin, realMax).intValue();
     }
+    private boolean isValidRange(int realMin, int realMax) {
```
Comment (snuyanzin) [inline src/main/java/net/datafaker/providers/base/Number.java:60]: missed empty line between methods

File: src/main/java/net/datafaker/providers/base/Barcode.java [hunk 6]
```diff
@@ -67,13 +67,7 @@ private long ean(int length) {
...
             odd = tmp;
         }
-        int var = 0;
-
-        if (length == 13) {
-            var = odd + even + (even << 1);
-        } else if (length == 8 || length == 14 || length == 12) {
-            var = odd + even + (odd << 1);
-        }
+        int var = calculateVar(length, odd, even);
```
[the remark this answers is not in the document]
Comment (kingthorin) [reply src/main/java/net/datafaker/providers/base/Barcode.java:70]: Do you have a linter for this, or is it just something that you pickup naturally?

File: src/main/java/net/datafaker/providers/base/Number.java (hunk 4 above)
Comment (almasfiza) [reply src/main/java/net/datafaker/providers/base/Number.java]: fixed

File: src/main/java/net/datafaker/providers/base/Number.java (hunk 5 above)
Comment (almasfiza) [reply src/main/java/net/datafaker/providers/base/Number.java]: > missed empty line between methods

fixed, thank you for helping me with good coding practices.

File: src/main/java/net/datafaker/providers/base/Barcode.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (almasfiza) [reply src/main/java/net/datafaker/providers/base/Barcode.java]: made the var final. 

> this should be `final`
> 
> the next 2 vars could be also made `final`

[the remark this answers is not in the document]
Comment (snuyanzin) [reply src/main/java/net/datafaker/providers/base/Barcode.java]: for this just naturally

**Comment by snuyanzin** [comment]
thanks for your contribution

lgtm, once ci is passed

**Comment by almasfiza** [comment]
> thanks for your contribution
> 
> lgtm, once ci is passed

All checks have passed.

**Review by kingthorin — APPROVED** [review]

**Pull request merged**